uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
+                shouldReject_DanglingRefCannotBeFixedAsOriginAlreadyExistsInAliasZone(IdentityZone.getUaa(), customZone);
             }
 
             @Test
-            void shouldReject_IdpWithOriginKeyAlreadyPresentInOtherZone_CustomToUaaZone() throws Exception {
-                shouldReject_IdpWithOriginKeyAlreadyPresentInOtherZone(customZone, IdentityZone.getUaa());
+            void shouldReject_DanglingRefCannotBeFixedAsOriginAlreadyExistsInAliasZone_CustomToUaaZone() throws Throwable {
+                shouldReject_DanglingRefCannotBeFixedAsOriginAlreadyExistsInAliasZone(customZone, IdentityZone.getUaa());
             }
 
-            private void shouldReject_IdpWithOriginKeyAlreadyPresentInOtherZone(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
-                // create IdP with origin key in zone 2
-                final IdentityProvider<?> existingIdpInZone2 = buildOidcIdpWithAliasProperties(zone2.getId(), null, null);
-                createIdp(zone2, existingIdpInZone2);
+            private void shouldReject_DanglingRefCannotBeFixedAsOriginAlreadyExistsInAliasZone(
+                    final IdentityZone zone1,
+                    final IdentityZone zone2
+            ) throws Throwable {
+                final IdentityProvider<?> existingIdp = executeWithTemporarilyEnabledAliasFeature(
+                        aliasFeatureEnabled,
+                        () -> createIdpWithAlias(zone1, zone2)
+                );
 
-                // create IdP with same origin key in zone 1
-                final IdentityProvider<?> idp = buildIdpWithAliasProperties(
-                        zone1.getId(),
-                        null,
+                // delete alias IdP and create a new one in zone 2 without alias but with the same origin
+                deleteIdpViaDb(existingIdp.getOriginKey(), zone2.getId());
+                final IdentityProvider<?> newIdpWithSameOrigin = buildOidcIdpWithAliasProperties(
+                        zone2.getId(),
                         null,
-                        existingIdpInZone2.getOriginKey(), // same origin key
-                        OIDC10
+                        null
                 );
-                final IdentityProvider<?> providerInZone1 = createIdp(zone1, idp);
+                newIdpWithSameOrigin.setOriginKey(existingIdp.getOriginKey());
+                createIdp(zone2, newIdpWithSameOrigin);
 
-                // update the alias ZID to zone 2, where an IdP with this origin already exists -> should fail
-                providerInZone1.setAliasZid(zone2.getId());
-                shouldRejectUpdate(zone1, providerInZone1, HttpStatus.CONFLICT);
+                existingIdp.setAliasId(null);
+                existingIdp.setAliasZid(null);
+                existingIdp.setName("some-new-name");
+                shouldRejectUpdate(zone1, existingIdp, HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.